Assets/MRTK/Core/Utilities/Editor/PackageManifest/PackageManifestUpdater.cs

```diff
@@ -182,44 +186,38 @@ internal static bool IsMSBuildForUnityEnabled()
             return false;
         }
 
+        [MenuItem("Mixed Reality Toolkit/MSBuild/Use MSBuild for Unity dependency resolution", true)]
+        private static bool ShowEnableMenuItem()
+        {
+            return !IsMSBuildForUnityEnabled();
+        }
+
+        [MenuItem("Mixed Reality Toolkit/MSBuild/Remove MSBuild for Unity", true)]
+        private static bool ShowDisableMenuItem()
+        {
+            return IsMSBuildForUnityEnabled();
+        }
+
         /// <summary>
         /// Ensures the required settings exist in the package manager to allow for using MSBuild for Unity.
         /// </summary>
-        internal static void EnsureMSBuildForUnity()
+        [MenuItem("Mixed Reality Toolkit/MSBuild/Use MSBuild for Unity dependency resolution", priority = 99)]
+        internal static void EnableMSBuildForUnity()
         {
-            PackageManifest manifest = null;
-
-            string manifestPath = GetPackageManifestFilePath();
-            if (string.IsNullOrWhiteSpace(manifestPath))
+            if (!EditorUtility.DisplayDialog(
+                "Use MSBuild for Unity for dependency resolution",
+                "Some versions of Unity have experienced issues with MSBuild for Unity. Do you wish to continue?",
```

Comment:
I was gonna leave a comment on the version that scoped this just to 2019+ but I know that this was just removed in a recent iteration.

I'm usually against dialogs like this - i.e. vague warnings that things might be busted. The next question would be - "why does this exist here if it has a non-zero probability that it's not gonna work?".

Also if they enable this on 2019, doesn't this get them into that permanent loop? i.e. how do they un-mess up their project?

Reply:
i have tested on recent 2019.3 builds and 2019.4 and have not seen the loop. the issue appears to have been with an early 2019.3 release.

good question on the "how do you fix"

Reply:
i cam happy remove the prompt as we recommend customers be on more recent 2019.3/4 builds